clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -278,9 +278,14 @@ void SYCL::fpga::BackendCompiler::ConstructJob(Compilation &C,
         if (Ty == types::TY_INVALID)
           continue;
         if (types::isSrcFile(Ty) || Ty == types::TY_Object) {
-          llvm::sys::path::replace_extension(AN, "prj");
-          ReportOptArg += Args.MakeArgString(AN);
-          break;
+          // Output is based on the filename only so strip off any directory
+          // information if provided with the source/object file.
+          AN = llvm::sys::path::filename(AN);
```

Comment:
As an out-of-scope note, `AN` is a tough name for figuring the meaning out - am I missing some common abbreviation?

Reply:
Another question on AN, or rather Args. Are those always filenames? Can't Args contain other options?

Reply:
`AN` was an abbreviation of Arg Name (string of the Arg).  I'll modify to be more succinct.  As for the Args question, the Args consist of all arguments passed on the command line.  So they are not all file names.  But good catch - there should be an additional check to make sure that the Arg we are currently looking at is an Input type.